packages/atomic/src/components/commerce/product-template-components/atomic-product-image/atomic-product-image.tsx

```diff
@@ -0,0 +1,153 @@
+import {Product, ProductTemplatesHelpers} from '@coveo/headless/commerce';
+import {Component, h, Prop, Element, State, Method} from '@stencil/core';
+import {Bindings} from '../../../../components';
+import {
+  InitializeBindings,
+  InitializableComponent,
+} from '../../../../utils/initialization-utils';
+import {filterProtocol} from '../../../../utils/xss-utils';
+import {Carousel} from '../../../common/carousel';
+import {ProductContext} from '../product-template-decorators';
+
+/**
+ * The `atomic-product-image` component renders an image from a product field.
+ */
+@Component({
+  tag: 'atomic-product-image',
+  styleUrl: 'atomic-product-image.pcss',
+  shadow: false,
+})
+export class AtomicProductImage implements InitializableComponent {
+  @InitializeBindings() public bindings!: Bindings;
+  @ProductContext() private product!: Product;
+  @Element() private host!: HTMLElement;
+  @State() private useFallback = false;
+  @State() private currentImage = 0;
+
+  /**
+   * The product field which the component should use. This will look for the field in the product object first, then in the product.additionalFields object.
+   */
+  @Prop({reflect: true}) field: string = 'ec_thumbnails';
+
+  /**
+   * An optional fallback image URL that will be used in case the specified image is not available or an error is encountered.
+   */
+  @Prop({reflect: true}) fallback?: string;
+
+  public error!: Error;
+
+  /**
+   * Moves to the previous image, when the carousel is activated.
+   */
+  @Method() public async previousImage() {
+    this.currentImage =
+      this.currentImage - 1 < 0
+        ? this.numberOfImages - 1
+        : this.currentImage - 1;
+  }
+
+  /**
+   * Moves to the next image, when the carousel is activated.
+   */
+  @Method() public async nextImage() {
+    this.currentImage = (this.currentImage + 1) % this.numberOfImages;
+  }
+
+  private logWarning(message: string) {
+    this.bindings.engine.logger.warn(message, this.host);
+  }
+
+  private handleImageError(url: string) {
+    const message = `The image url "${url}" is not valid or could not be loaded. You might want to add a "fallback" property.`;
+
+    this.fallback ? (this.useFallback = true) : this.logWarning(message);
+  }
+
+  private get imageUrls() {
+    const value = ProductTemplatesHelpers.getProductProperty(
+      this.product,
+      this.field
+    );
+
+    return Array.isArray(value) ? value : [value];
+  }
+
```

Comment:
All products in Barca have only one image. If you want to test the carousel you can do something like that:

```typescript
  private get imageUrls() {
    const value = ProductTemplatesHelpers.getProductProperty(
      this.product,
      this.field
    );

    let images = Array.isArray(value) ? value : [value];

    images = [
      ...images,
      'https://placehold.co/600/orange/white',
      'https://placehold.co/600/red/white',
      'https://placehold.co/600/blue/white',
    ];

    return images;
  }
```